terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -270,11 +270,7 @@ def get_proxied_source(raw_source) # rubocop:disable Metrics/AbcSize
         url = raw_source.split(%r{(?<!:)//}).first + "?terraform-get=1"
         host = URI.parse(raw_source).host
 
-        response = Excon.get(
-          url,
-          idempotent: true,
-          **SharedHelpers.excon_defaults
-        )
+        response = Dependabot::RegistryClient.get(url: url)
```

Comment:
I found it curious that the file parser doesn't already use `Dependabot::Terraform::RegistryClient`, but at a cursory read it seems like it is always going to be making unauthenticated calls so this is likely working as intended.

@jurre am I correct in thinking we intentionally do not use the terraform client here?

Reply:
It's been a minute since I've worked on this, but I do think you're right yes